layer/runtime_layer_data.cc

```diff
@@ -131,9 +135,20 @@ void RuntimeLayerData::LogAndRemoveQueryPools() {
       // FIXME: We should adjust the elapsed units to account for the current
       // GPU frequency. Calling vkGetPhysicalDeviceProperties here causes the
       // driver to crash, however.
-      Log("pipeline_execution", GetPipelineHash(info->pipeline),
-          absl::StrCat(timestamp1 - timestamp0, ",", invocations[0], ",",
-                       invocations[1]));
+      HashVector pipeline = GetPipelineHash(info->pipeline);
+      std::vector<int64_t> hashes(pipeline.begin(), pipeline.end());
+      RuntimeEvent event("pipeline_execution", hashes,
+                         DurationClock::duration(timestamp1 - timestamp0),
```

Comment:
Does this assume nanosecond-level precision? Would it be possible to call a function that explicitly constructs duration with some unit?

Reply:
Yes, that sounds great!

Reply:
Or you could add your own class that wraps `DurationClock::duration` and has a static function to construct from nanoseconds, e.g.:
```c++
class Duration {
public:
  static Duration FromNanoseconds(int64_t nanos) { ... };

private:
  DurationClock::duration duration_;
};
```

Reply:
This looks cleaner.

> Or you could add your own class that wraps `DurationClock::duration` and has a static function to construct from nanoseconds, e.g.:
> 
> ```c++
> class Duration {
> public:
>   static Duration FromNanoseconds(int64_t nanos) { ... };
> 
> private:
>   DurationClock::duration duration_;
> };
> ```

This looks cleaner. Thanks!